Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -2448,6 +2462,42 @@ static NSStringCompareOptions compareOptionsForFindOptions(WebCore::FindOptions
     });
 }
 
+RefPtr<TextIndicator> UnifiedPDFPlugin::textIndicatorForSelection(OptionSet<WebCore::TextIndicatorOption> options, WebCore::TextIndicatorPresentationTransition transition)
+{
+    if (auto selectionBounds = selectionBoundsForFirstPageInDocumentSpace(m_currentSelection)) {
+        auto rectInDocumentCoordinates = *selectionBounds;
+        auto rectInContentsCoordinates = convertFromDocumentToContents(rectInDocumentCoordinates);
+        auto rectInRootViewCoordinates = convertFromPluginToRootView(convertFromDocumentToPlugin(rectInDocumentCoordinates));
+
+        float deviceScaleFactor = this->deviceScaleFactor();
+
+        auto buffer = ImageBuffer::create(rectInRootViewCoordinates.size(), RenderingPurpose::ShareableSnapshot, deviceScaleFactor, DestinationColorSpace::SRGB(), PixelFormat::BGRA8, { }, nullptr);
+        if (!buffer)
+            return nullptr;
+
+        auto& context = buffer->context();
+        context.translate(-rectInContentsCoordinates.location());
+        context.scale(m_scaleFactor);
+        context.translate(centeringOffset());
```

Comment:
I'm a bit confused by this. Would be nicer of when `paintPDFContent()` is called the context is always set up with the coordinate system that matches `m_contentsLayer` painting.

Reply:
That's what I'm doing, right? (but also shifting it so the origin is at the rect we're drawing)